pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java

```diff
@@ -260,10 +276,18 @@ protected BrokerResponse handleRequest(long requestId, String query, @Nullable S
     brokerResponse.setPartialResult(isPartialResult(brokerResponse));
 
     // Set total query processing time
-    // TODO: Currently we don't emit metric for QUERY_TOTAL_TIME_MS
     long totalTimeMs = TimeUnit.NANOSECONDS.toMillis(
         sqlNodeAndOptions.getParseTimeNs() + (executionEndTimeNs - compilationStartTimeNs));
     brokerResponse.setTimeUsedMs(totalTimeMs);
+    if (brokerResponse.isNumGroupsLimitReached()) {
+      for (String tableName : tableNames) {
+        _brokerMetrics.addMeteredTableValue(tableName, BrokerMeter.BROKER_RESPONSES_WITH_NUM_GROUPS_LIMIT_REACHED, 1);
+      }
```

Comment:
We were planning to do something similar before and had some reservations around this. More info on this thread -- https://github.com/apache/pinot/pull/11023#discussion_r1248823295
cc @ankitsultana